pkg/controller/route/labels.go

```diff
@@ -103,13 +115,34 @@ func (c *Controller) deleteLabelForOutsideOfGivenConfigurations(
 	// Delete label for newly removed configurations as traffic target.
 	for _, config := range oldConfigsList {
 		if _, ok := configMap[config.Name]; !ok {
+
 			delete(config.Labels, serving.RouteLabelKey)
-			if _, err := c.ServingClientSet.ServingV1alpha1().Configurations(ns).Update(config); err != nil {
-				logger.Errorf("Failed to update Configuration %s: %s", config.Name, err)
+
+			configClient := c.ServingClientSet.ServingV1alpha1().Configurations(config.Namespace)
+			if err := setRouteLabelForConfiguration(configClient, config.Name, nil); err != nil {
+				logger.Errorf("Failed to remove route label to configuration %q: %s", config.Name, err)
 				return err
 			}
 		}
 	}
 
 	return nil
 }
+
+func setRouteLabelForConfiguration(configClient servingv1alpha1.ConfigurationInterface, configName string, routeName *string) error {
+	mergePatch := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": map[string]interface{}{
+				serving.RouteLabelKey: routeName,
```

Comment:
seems weird to make this `*string` instead of just passing the string and taking the `&routeName` here.

Reply:
You need to be able to have null in the json so that it deletes the keyed entry from the object

Reply:
I'll add a comment because it's not very clear 

for context: https://tools.ietf.org/html/rfc7386#section-1